src/components/dashboard/forms/items/NewItemFormTags.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import actions from 'actions'
+import NewItemHoc from './NewItemHocStep'
+import Translate from 'components/translate/Translate'
+import Grid from '@material-ui/core/Grid'
+import Autocomplete from 'components/common/autocomplete'
+import { getTags } from 'services/adex-node/actions'
+
+const noTagsErrMsg = 'ERR_REQUIRED_FIELD'
+
+class NewItemFormTags extends Component {
+    componentWillMount() {
+        getTags({authSig: this.props.account._authSig})
+            .then((res) => {
+                const tags = res.reduce((o, key) => ({ ...o, [key._id]: key._id}), {})
+
+                this.props.actions.updateTags({tags: tags})
+            })
+        this.props.validate('tags', {isValid: false, err: {msg: noTagsErrMsg}})                             
+    }
+
+    addTagsForDisplay(tags) {
+        if (!tags) {
+            return null
+        }
+        tags.forEach(tag => {
+            if (this.doesTagExist(tag)) {
+                return null
+            }
+    
+            this.props.actions.addNewTag({tag: tag})
+        })
+    }
+
+    doesTagExist(tag) {
+        return tag in this.props.tags
+    }
+
+    render() {
+        // Makes sure if you go one step forward and then back the unsaved tags still appear
+        this.addTagsForDisplay(this.props.item.meta.tags)
```

Comment:
Don't need this here. Actually the proper way is to call this.props.actions.addNewTag({tag: tag}) on autocomplete change. This should be enough. You don't need this function at all.